### ISSUE TITLE
Can anyone fix my problem, Im new and beef will not open :/

### ISSUE THREAD

**Issue by Social7451** [beefproject/beef#1603]
$ cd /usr/share/beef-xss/
ethan@kali2018:/usr/share/beef-xss$ ./beef
[ 1:34:13][*] Bind socket [imapeudora1] listening on [0.0.0.0:2000].
[ 1:34:13][*] Browser Exploitation Framework (BeEF) 0.4.7.0-alpha
[ 1:34:13]    |   Twit: @beefproject
[ 1:34:13]    |   Site: http://beefproject.com
[ 1:34:13]    |   Blog: http://blog.beefproject.com
[ 1:34:13]    |_  Wiki: https://github.com/beefproject/beef/wiki
[ 1:34:13][*] Project Creator: Wade Alcorn (@WadeAlcorn)
Traceback (most recent call last):
	26: from ./beef:98:in `<main>'
	25: from /usr/lib/ruby/vendor_ruby/dm-migrations/auto_migration.rb:27:in `auto_upgrade!'
	24: from /usr/lib/ruby/vendor_ruby/dm-migrations/auto_migration.rb:46:in `repository_execute'
	23: from /usr/lib/ruby/vendor_ruby/dm-core/support/descendant_set.rb:63:in `each'
	22: from /usr/lib/ruby/vendor_ruby/dm-core/support/subject_set.rb:211:in `each'
	21: from /usr/lib/ruby/vendor_ruby/dm-core/support/ordered_set.rb:320:in `each'
	20: from /usr/lib/ruby/vendor_ruby/dm-core/support/ordered_set.rb:320:in `each'
	19: from /usr/lib/ruby/vendor_ruby/dm-core/support/ordered_set.rb:320:in `block in each'
	18: from /usr/lib/ruby/vendor_ruby/dm-core/support/subject_set.rb:211:in `block in each'
	17: from /usr/lib/ruby/vendor_ruby/dm-core/support/descendant_set.rb:64:in `block in each'
	16: from /usr/lib/ruby/vendor_ruby/dm-migrations/auto_migration.rb:47:in `block in repository_execute'
	15: from /usr/lib/ruby/vendor_ruby/dm-migrations/auto_migration.rb:145:in `auto_upgrade!'
	14: from /usr/lib/ruby/vendor_ruby/dm-migrations/auto_migration.rb:73:in `upgrade_model_storage'
	13: from /usr/lib/ruby/vendor_ruby/dm-migrations/adapters/dm-do-adapter.rb:57:in `upgrade_model_storage'
	12: from /usr/lib/ruby/vendor_ruby/dm-migrations/adapters/dm-do-adapter.rb:90:in `create_model_storage'
	11: from /usr/lib/ruby/vendor_ruby/dm-migrations/adapters/dm-sqlite-adapter.rb:18:in `storage_exists?'
	10: from /usr/lib/ruby/vendor_ruby/dm-migrations/adapters/dm-sqlite-adapter.rb:43:in `table_info'
	 9: from /usr/lib/ruby/vendor_ruby/dm-do-adapter/adapter.rb:33:in `select'
	 8: from /usr/lib/ruby/vendor_ruby/dm-do-adapter/adapter.rb:276:in `with_connection'
	 7: from /usr/lib/ruby/vendor_ruby/dm-do-adapter/adapter.rb:251:in `open_connection'
	 6: from /usr/lib/ruby/vendor_ruby/data_objects/connection.rb:68:in `new'
	 5: from /usr/lib/ruby/vendor_ruby/data_objects/pooling.rb:121:in `new'
	 4: from /usr/lib/ruby/vendor_ruby/data_objects/pooling.rb:176:in `new'
	 3: from /usr/lib/ruby/vendor_ruby/data_objects/pooling.rb:176:in `synchronize'
	 2: from /usr/lib/ruby/vendor_ruby/data_objects/pooling.rb:181:in `block in new'
	 1: from /usr/lib/ruby/vendor_ruby/data_objects/connection.rb:79:in `new'
/usr/lib/ruby/vendor_ruby/data_objects/connection.rb:79:in `initialize': unable to open database file (DataObjects::ConnectionError)

**Comment by passbe** [comment]
@Social7451 is this a fresh copy of beef? Do you see a /beef.db?

**Comment by noraj** [comment]
Hi @passbe 

I installed BeEF using Blackarch repo. (source [here](https://github.com/BlackArch/blackarch/tree/master/packages/beef)) and I have the same error.

## The error

```
$ beef                                                                                         
[15:05:04][*] Browser Exploitation Framework (BeEF) 0.4.7.0-alpha                                                                          
[15:05:04]    |   Twit: @beefproject                                                                                                       
[15:05:04]    |   Site: https://beefproject.com                                                                                            
[15:05:04]    |   Blog: http://blog.beefproject.com                                                                                        
[15:05:04]    |_  Wiki: https://github.com/beefproject/beef/wiki                                                                           
[15:05:04][*] Project Creator: Wade Alcorn (@WadeAlcorn)                                                                                   
Traceback (most recent call last):                                                                                                         
        26: from beef:138:in `<main>'                                                                                                      
        25: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/auto_migration.rb:27:in `auto_upgrade!
'                                                                                                                                          
        24: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/auto_migration.rb:46:in `repository_ex
ecute'                                                                                                                                     
        23: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-core-1.2.1/lib/dm-core/support/descendant_set.rb:63:in `each'           
        22: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-core-1.2.1/lib/dm-core/support/subject_set.rb:211:in `each'             
        21: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-core-1.2.1/lib/dm-core/support/ordered_set.rb:320:in `each'             
        20: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-core-1.2.1/lib/dm-core/support/ordered_set.rb:320:in `each'             
        19: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-core-1.2.1/lib/dm-core/support/ordered_set.rb:320:in `block in each'    
        18: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-core-1.2.1/lib/dm-core/support/subject_set.rb:211:in `block in each'    
        17: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-core-1.2.1/lib/dm-core/support/descendant_set.rb:64:in `block in each'  
        16: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/auto_migration.rb:47:in `block in repository_execute'
        15: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/auto_migration.rb:145:in `auto_upgrade!'
        14: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/auto_migration.rb:73:in `upgrade_model_storage'
        13: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/adapters/dm-do-adapter.rb:57:in `upgrade_model_storage'
        12: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/adapters/dm-do-adapter.rb:90:in `create_model_storage'
        11: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/adapters/dm-sqlite-adapter.rb:18:in `storage_exists?'
        10: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-migrations-1.2.0/lib/dm-migrations/adapters/dm-sqlite-adapter.rb:43:in `table_info'
         9: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-do-adapter-1.2.0/lib/dm-do-adapter/adapter.rb:33:in `select'            
         8: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-do-adapter-1.2.0/lib/dm-do-adapter/adapter.rb:276:in `with_connection'  
         7: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/dm-do-adapter-1.2.0/lib/dm-do-adapter/adapter.rb:251:in `open_connection'  
         6: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/data_objects-0.10.17/lib/data_objects/connection.rb:68:in `new'            
         5: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/data_objects-0.10.17/lib/data_objects/pooling.rb:121:in `new'              
         4: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/data_objects-0.10.17/lib/data_objects/pooling.rb:176:in `new'              
         3: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/data_objects-0.10.17/lib/data_objects/pooling.rb:176:in `synchronize'      
         2: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/data_objects-0.10.17/lib/data_objects/pooling.rb:181:in `block in new'     
         1: from /usr/share/beef/vendor/bundle/ruby/2.5.0/gems/data_objects-0.10.17/lib/data_objects/connection.rb:79:in `new'            
/usr/share/beef/vendor/bundle/ruby/2.5.0/gems/data_objects-0.10.17/lib/data_objects/connection.rb:79:in `initialize': unable to open database file (DataObjects::ConnectionError)
```

## Info

```
$ ruby --version 
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]

$ sqlite3 --version
3.25.2 2018-09-25 19:08:10 fb90e7189ae6d62e77ba3a308ca5d683f90bbe633cf681865365b8e92792d1c7

$ node --version
v8.12.0
```

## Troubleshooting

I don't see any `beef.db`.

```
$ grep -ri beef\.db /usr/share/beef/
/usr/share/beef/config.yaml:        db_file: "beef.db"
```

**Comment by noraj** [comment]
**Update** Beef was requiring root permissions to write `beef.db` in `/usr/share/beef/`.

**Comment by bcoles** [comment]
Thanks @noraj

Looks like BeEF should be modified to rescue `unable to open database file (DataObjects::ConnectionError)` and display a more meaningful error message.

**Comment by bcoles** [comment]
This should be fixed in cb2e8b66818914828937f8df9c79c4759e5033d0.

BeEF now rescues from failed database reset (ie, if the `-x` flag was parsed).

If the connection fails, and the `-x` flag wasn't passed, then no error message will be thrown and BeEF will continue to appear that it's working, when it is in fact broken, and no zombies will be able to connect. This should probably be fixed one day. Today is not that day.

Closing this issue.
